src/lib/string.ts

```diff
@@ -45,3 +45,109 @@ export function uncapitalize(value?: string): string | undefined {
 
   return value.charAt(0).toLowerCase() + value.slice(1);
 }
+
+/**
+ * Truncates a string to a maximum length, adding a suffix if truncated
+ * @param value The string to truncate
+ * @param maxLength The maximum length of the resulting string
+ * @param suffix The suffix to append if truncated (default: "...")
+ * @returns The truncated string
+ */
+export function truncate(value: string | null | undefined, maxLength: number, suffix = "..."): string {

```

Comment:
the value parameter should be like in the existing functions
```suggestion
export function truncate(value?: string, maxLength: number, suffix = "..."): string {
```